lib/hanrahashiv.py

```diff
@@ -0,0 +1,84 @@
+from manim import Tex, MathTex
+from manim import UP, DOWN, LEFT, DL
+from manim import VGroup
+from manim import Scene
+from manim import ReplacementTransform
+
+class FormulaModifications(Scene):
+
+# FIXME մարդավարի docstring գրել ամեն ֆունկցիայի համար
+#  միգուցե էս ամենը տեղափոխել QarakusiScene (որտև մենակ միանդամների մեջ չի, որ պետք կգան)
+    """
+        new_sequence addresses each of formula's item to it's new position, including '$\cdot$', ' + '
+        to work properly, formula must be written in such a way that every item is in different apostrophes like so
+        formula = Tex('a', '$\cdot$', 'b')
+    """
+
+    def rearrange_formula(self, formula : Tex or MathTex, new_sequence : list, move_up : list, move_down : list):
+        tex_string_list = [tex.get_tex_string() for tex in formula]
+        new_tex_string_list = [tex_string_list[new_sequence[i]] for i in range(len(tex_string_list))]
+        new_formula = Tex(*new_tex_string_list, font_size=formula[0].font_size, color=formula.color).move_to(formula)
+
+        in_line = []
+        for i in range(len(tex_string_list)):
+            if i not in move_up and i not in move_down:
+                in_line.append(i)
+
+        self.play(
+            VGroup(*[formula[i] for i in move_up]).animate.shift(0.5 * UP),
+            VGroup(*[formula[i] for i in move_down]).animate.shift(0.5 * DOWN)
```

Comment:
Please, don't use `VGroup` if you don't need it, I think `manim` does a lot of work at initialization that you don't need. And it's not obvious.

You can do something like:
```suggestion
            *[formula[i].animate.shift(0.5 * UP) for i in move_up]),
            *[formula[i].animate.shift(0.5 * DOWN) for i in move_down])
```